### PR TITLE
Improve headbugfix

### PR DESCRIPTION
**Pull request by ViperMaul**
Disable headbugfix for Handcuffed/Surrender,unconscious,water surface and log the situation.

### CODE REVIEW

File: addons/common/functions/fnc_headBugFix.sqf [hunk 1]
```diff
@@ -10,21 +10,30 @@
...
+_dir = getDir ACE_player;
+
 titleCut ["", "BLACK"];
-_pos = getposATL player;
-_dir = getDir player;
-_anim = animationState player;
+
 // create invisible headbug fix vehicle
-_ACE_HeadbugFix = createVehicle ["ACE_Headbug_Fix", getposATL player, [], 0, "NONE"];
+_ACE_HeadbugFix = createVehicle ["ACE_Headbug_Fix", [0,0,10000], [], 0, "NONE"];
```
Comment (SilentSpike) [inline addons/common/functions/fnc_headBugFix.sqf]: Is it possible to do this with only a local vehicle? (Just to save on the network traffic)

**Comment by gienkov** [comment]
There was a timeout implemented in ACE2 headbug fix, I don't see it in ACE3, you should add ~10 sec timeout before you can use headbug fix.

File: addons/common/functions/fnc_headBugFix.sqf (hunk 1 above)
Comment (PabstMirror) [inline addons/common/functions/fnc_headBugFix.sqf]: I think this is rare enough, the traffic shouldn't be that bad.  Also, I'm not sure if this is an issue, but it's important the animation is reset on all clients and a local vehicle might?? not sync correctly.

Minor, but moving the player to [0,0,10000] could cause issues with mission triggers (eg blufor not present), what about just
`(getposATL ACE_player) vectorAdd [0,0,10000]`

Comment (Glowbal) [inline addons/common/functions/fnc_headBugFix.sqf]: Can we hide the player object instead of teleporting it away?

**Comment by alganthe** [comment]
The transition is used as timeout, :+1: @ViperMaul

File: addons/common/functions/fnc_headBugFix.sqf (hunk 1 above)
Comment (SilentSpike) [inline addons/common/functions/fnc_headBugFix.sqf]: A framework was added to hide units: `EFUNC(common,hideUnit)` and `EFUNC(common,unhideUnit)`

**Comment by ViperMaul** [comment]
@gienkov show me the code you are talking about. I grabbed the code from ACE2 already.

**Comment by ViperMaul** [comment]
Thanks guys those comments helped make it look much nicer in my dedi tests.

**Comment by Glowbal** [comment]
Is this ready?

**Comment by ViperMaul** [comment]
Yes!

**Pull request merged**